component/src/main/java/org/wso2/extension/siddhi/io/feed/sink/FeedSink.java

```diff
@@ -0,0 +1,268 @@
+                || atomFunc.equals(Constants.FEED_DELETE) || atomFunc.equals(Constants.FEED_CREATE)) {
+            return atomFunc;
+        }
+        throw new SiddhiAppValidationException("Atom function validation error  in " + streamDefinition.getId()
+                + ". Acceptance parameters are 'create', 'delete', 'update'");
+    }
+
+    private int validateTimeOut() {
+        try {
+            int timeout = Integer.parseInt(optionHolder.validateAndGetStaticValue(Constants.TIME_OUT,
+                    Constants.DEFAULT_TIME_OUT));
+            if (timeout > 0) {
+                return timeout;
+            } else {
+                throw new SiddhiAppValidationException("Error in " + streamDefinition.getId()
+                        + " validating timeout, Response timeout accept only positive integers. But found " +
+                        optionHolder.validateAndGetStaticValue(Constants.TIME_OUT,
+                                Constants.DEFAULT_TIME_OUT));
+            }
+        } catch (NumberFormatException e) {
+            throw new SiddhiAppValidationException("Error in " + streamDefinition.getId()
+                    + " validating timeout, Response timeout accept only positive integers. But found " +
+                    optionHolder.validateAndGetStaticValue(Constants.TIME_OUT,
+                            Constants.DEFAULT_TIME_OUT));
+        }
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions dynamicOptions) throws ConnectionUnavailableException {
+        HashMap<String, String> map = (HashMap) payload;
+        ClientResponse resp = null;
+        switch (atomFunc) {
+            case Constants.FEED_CREATE: {
+                Entry entry = EntryUtils.createEntry(map, abdera.newEntry());
+                entry.setPublished(new Date());
+                try {
+                    resp = abderaClient.post(url.toString(), entry);
+                } catch (RuntimeException exception) {
+                    throw new FeedErrorResponseException("Connection timeout exception in " + streamDefinition.getId()
+                            + ". The host did not accept the connection within timeout of "
+                            + abderaClient.getConnectionTimeout());
+                }
+                break;
+            }
+            case Constants.FEED_DELETE: {
+                try {
+                    resp = abderaClient.delete(map.get("id"));
+                } catch (RuntimeException exception) {
+                    throw new FeedErrorResponseException("Connection timeout exception in " + streamDefinition.getId()
+                            + ". The host did not accept the connection within timeout of "
+                            + abderaClient.getConnectionTimeout());
+                }
+                break;
+            }
+            case Constants.FEED_UPDATE: {
+                try {
+                    resp = abderaClient.get(url.toString());
```

Comment:
Why can't we create a single connection form `abderaClient.get(url.toString())` in the connect method and use it in the publish? Because the **atomFunc** doesn't even change dynamically